src/Controller/Entry/EntryFrontController.php

```diff
@@ -17,46 +16,62 @@
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\Security\Http\Attribute\IsGranted;
 
 class EntryFrontController extends AbstractController
 {
     public function __construct(private readonly EntryRepository $repository)
     {
     }
 
-    public function root(?string $sortBy, ?string $time, ?string $type, Request $request): Response
+    public function root(?string $sortBy, ?string $time, ?string $type, string $federation, Request $request): Response
     {
-        $user = $this->getUser();
-
-        if (!$user) {
-            return $this->front($sortBy, $time, $type, $request);
-        }
-
-        $front = match ($user->homepage) {
-            User::HOMEPAGE_SUB => 'subscribed',
-            User::HOMEPAGE_MOD => 'moderated',
-            User::HOMEPAGE_FAV => 'favourite',
-            default => 'front',
-        };
-
-        return $this->$front($sortBy, $time, $type, $request);
+        return $this->front($sortBy, $time, $type, $request->query->get('filter'), $federation, $request);
     }
 
-    public function front(?string $sortBy, ?string $time, ?string $type, Request $request): Response
+    public function front(?string $sortBy,
+        ?string $time,
+        ?string $type,
+        ?string $filter,
+        string $federation,
+        Request $request): Response
     {
         $user = $this->getUser();
+
+        if (null === $filter) {
+            if (null !== $request->query->get('filter')) {
+                $filter = $request->query->get('filter');
+            } elseif ($user) {
```

Comment:
this should not be an `elseif` should it?

Reply:
I think it should be -- my feeling on the logic was:

* If the caller of the function has requested a particular filter, use that (that happens if there's one at the front of the URL)
* Otherwise if there's a filter query parameter, use that
* Otherwise if there's a logged-in user, use their default
* Otherwise use 'all'

That sounds right, right? If so, maybe it'd be a little clearer as:

```
if ($filter) {
    // nothing to do
} elseif (null !== $request->query->get('filter')) {
    $filter = $request->query->get('filter');
} elseif ($user) {
    $filter = match ($user->homepage) {
                    User::HOMEPAGE_SUB => 'sub',
                    User::HOMEPAGE_MOD => 'mod',
                    User::HOMEPAGE_FAV => 'fav',
                    default => 'all',
                };
} else {
   $filter = 'all';
}
```

?

Also, the handling of $filter in general is a little messed up; see my comment on the fav/sub/mod diff.

Reply:
checks out :+1:

Reply:
Should be mooted now, by the single-route changes